### DISCUSSION TITLE
v4.5.1

### DISCUSSION THREAD

**Discussion by fklassen** [appneta/tcpreplay#893] [Announcements]
## Announcing v4.5.1

This release contains contributions from a record number of new contributors. This is greatly appreciated since I am a team of one, and do Tcpreplay maintenance in my spare time.

There are many bug fixes and new features. Most notable features:

* AF_XDP socket support - if you have a newer Linux kernel, you will be able to transmit at line rates without having to install 3rd party kernel modules (e.g. netmap, PF_RING)
* `-w` tcpreplay option - this overrides the `-i` option, and allows you to write to a PCAP file rather than an interface
* `--include` and `--exclude` tcpreplay options - allows replay of a list of specific packet numbers to replay. This may slow things down, so consider using in combination with `-w`.
* `--fixhdrlen` tcpreplay option - added to control action on packet length changes
* `-W` tcpreplay option - suppress warnings when replaying
* SLL2( Linux "cooked" capture encapsulation v2)
* Haiku support

## What's Changed
* Add support for LINUX_SLL2 by @btriller in https://github.com/appneta/tcpreplay/pull/728
* Feature #727 - Linux SLL v2 by @fklassen in https://github.com/appneta/tcpreplay/pull/820
* Bug #779 - honour overflow for all PPS values by @fklassen in https://github.com/appneta/tcpreplay/pull/821
* AF_XDP socket extension using libxdp api by @plangarbalint in https://github.com/appneta/tcpreplay/pull/797
* Feature #822 -  AF_XDP socket extension by @fklassen in https://github.com/appneta/tcpreplay/pull/823
* Nanosec accurate packet processing by @plangarbalint in https://github.com/appneta/tcpreplay/pull/796
* Handle IPv6 fragment extension header by @ChuckCottrill in https://github.com/appneta/tcpreplay/pull/832
* Bug #837 - handle IPv6 fragment extension header by @fklassen in https://github.com/appneta/tcpreplay/pull/838
* Feature #796 - nanosecond packet processing by @fklassen in https://github.com/appneta/tcpreplay/pull/836
* configure.ac: unify search dirs for pcap and add lib32 by @shr-project in https://github.com/appneta/tcpreplay/pull/819
* Feature #839 - add pull request template by @fklassen in https://github.com/appneta/tcpreplay/pull/840
* ipv6 - add check for extension header length by @GabrielGanne in https://github.com/appneta/tcpreplay/pull/842
* Bug #827 PR #842 IPv6 extension header -  staging by @fklassen in https://github.com/appneta/tcpreplay/pull/859
* add check for empty cidr by @GabrielGanne in https://github.com/appneta/tcpreplay/pull/843
* Bug #824 and PR #843: check for empty CIDR by @fklassen in https://github.com/appneta/tcpreplay/pull/860
* Add option to turn on/off fix packet header length by @ChuckCottrill in https://github.com/appneta/tcpreplay/pull/846
* Bug #703 #844 PR #846: optionally fix packet header length --fixhdrlen by @fklassen in https://github.com/appneta/tcpreplay/pull/861
* Bug 863: fix nansecond timestamp regression by @fklassen in https://github.com/appneta/tcpreplay/pull/865
* autotools - AC_HELP_STRING is obsolete in 2.70 by @GabrielGanne in https://github.com/appneta/tcpreplay/pull/856
* some Haiku support by @infrastation in https://github.com/appneta/tcpreplay/pull/847
* configure.ac: do not run conftest in case of cross compilation by @ChenQi1989 in https://github.com/appneta/tcpreplay/pull/849
* dlt_jnpr_ether_cleanup: check config before cleanup by @Marsman1996 in https://github.com/appneta/tcpreplay/pull/851
* Fix recursive tcpedit cleanup by @GabrielGanne in https://github.com/appneta/tcpreplay/pull/855
* Bug #813: back out PR #855 by @fklassen in https://github.com/appneta/tcpreplay/pull/866
* Bug #867 - run regfree() on close by @fklassen in https://github.com/appneta/tcpreplay/pull/868
* Bug #869 tcpprep memory leak include exclude by @fklassen in https://github.com/appneta/tcpreplay/pull/870
* Bug #811 - add check for invalid jnpr header length by @fklassen in https://github.com/appneta/tcpreplay/pull/872
* Bug #792 avoid assertion and other fixes by @fklassen in https://github.com/appneta/tcpreplay/pull/873
* Bug #844 tap: ignore TUNSETIFF EBUSY errors by @fklassen in https://github.com/appneta/tcpreplay/pull/874
* Bug #876 - add missing free_umem_and_xsk function by @fklassen in https://github.com/appneta/tcpreplay/pull/877
* Feature #878 - add -w / --suppress-warning option by @fklassen in https://github.com/appneta/tcpreplay/pull/879
* Bug #835 false unsupported dlt warnings on 802.3 (Ethernet I) and LLC by @fklassen in https://github.com/appneta/tcpreplay/pull/880
* Feature #884 include exclude options by @fklassen in https://github.com/appneta/tcpreplay/pull/885
* Feature #853 direct traffic to pcap by @fklassen in https://github.com/appneta/tcpreplay/pull/871
* Feature #853 restore missing -P command by @fklassen in https://github.com/appneta/tcpreplay/pull/887
* Bug #888: check for map == NULL in cidr.c by @fklassen in https://github.com/appneta/tcpreplay/pull/889

## New Contributors
* @btriller made their first contribution in https://github.com/appneta/tcpreplay/pull/728
* @plangarbalint made their first contribution in https://github.com/appneta/tcpreplay/pull/797
* @ChuckCottrill made their first contribution in https://github.com/appneta/tcpreplay/pull/832
* @shr-project made their first contribution in https://github.com/appneta/tcpreplay/pull/819
* @infrastation made their first contribution in https://github.com/appneta/tcpreplay/pull/847
* @ChenQi1989 made their first contribution in https://github.com/appneta/tcpreplay/pull/849

**Full Changelog**: https://github.com/appneta/tcpreplay/compare/v4.4.4...v4.5.1

<hr /><em>This discussion was created from the release <a href='https://github.com/appneta/tcpreplay/releases/tag/v4.5.1'>v4.5.1</a>.</em>

**Comment by bsmojver** [comment]
Hi @fklassen,

Has CVE-2024-3024 been fixed in this release?

**Reply by fklassen** [reply]
I've never seen that one. I only see CVE's if they are added to this repo's Issues. It looks like a dupe and is probably fixed, but it needs verification. I'm back to my regular job, so it will be a bit before I have time to verify.

**Reply by bsmojver** [reply]
I created a binary file from hex dump in the CVE report and ran `tcpreplay` against it (both the patched 4.4.4 version in Fedora and 4.5.1) and I got this:
```
$ tcpprep --port --pcap=poc --cachefile=/dev/null

Fatal Error: Error opening libpcap: unknown file format
```

So, it appears to be fixed, just like you said.

**Reply by fklassen** [reply]
Thanks for doing the leg work for me.
